draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -460,6 +460,70 @@ Set-Cookie: lang=; Expires=Sun, 06 Nov 1994 08:49:37 GMT
 Cookie: SID=31d4d96e407aad42
 ~~~
 
+## Which Requirements to Implement
+
+This section helps guide implementors in determining which requirements and
+syntax they should implement. Choosing the wrong set of requirements could
+result in a lack of compatibility with other cookie implementations.
+
+It's important to note that being compatible means different things
+depending on the implementor's goals. These differences have built up over time
+due to both intentional and unintentional spec changes, spec interpretations,
+and historical implementation differences.
+
+### Cookie Producing Implementations
+
+An implementor should choose {{sane-profile}} whenever cookies are created and
+will be sent to a user agent, such as a web browser. These implementations are
+frequently referred to as Servers by the spec but that term includes anything
+which primarily produces cookies. Some potential examples:
+
+* Server applications hosting a website
+
+* Programming languages or software frameworks that supports cookies
+
+* Website add-ons, such as a business management suite
+
+All these benefit from not only supporting as many user agents as possible but
+also supporting other servers. This is useful if a cookie is produced by a
+software framework and is later sent back to a server application which needs
+to read it. {{sane-profile}} advises best practices that help maximize this
+sense of compatibility.
+
+See the latter section for more details on programming languages and software
+frameworks.
+
+### Cookies Consuming Implementations
+
+An implementor should choose {{ua-requirements}} whenever cookies are primarily
+received from another source. These implementations are referred to as user
+agents. Some examples:
+
+* Web browsers
+
+* Tools that support stateful http
+
+* Programming languages or software frameworks that support cookies
+
+Because user agents don't know which servers a user will access, and whether
+or not that server is following best practices, users agents are advised to
+implement a more lenient set of requirements and to accept some things that
+servers are warned against producing. {{ua-requirements}} advises best
+practices that help maximize this sense of compatibility.
+
+See the latter section for more details on programming languages and software
+frameworks.
+
+#### Programming Languages & Software Frameworks
+
+A programming language or software framework with support for cookies could
+reasonably be used as a cookie producer, cookie consumer, or both. Because
+a user of these may want to maximize their compatibility as either a producer
+or consumer these languages or frameworks should strongly consider supporting
```

Comment:
Context: Problems have occurred because languages allowed production of cookies that servers couldn't parse (because those servers implemented the more restricted requirements).

What I'm advising here is that languages/frameworks should by default only allow production of cookies that match the more restrictive, server, requirements. The more lenient ua requirements should be opt-in only. 

This'll hopefully make it harder for users to create cookies that won't be accepted elsewhere, but will still allow a user to write their UA (or anything that needs to consume potentially non-conformant cookies)

Reply:
Yeah, understood on the context. I think the sentence that follows explains what you mean, but the "also" implies supporting both sets of requirements is kinda different. Maybe just remove "also"?

Reply:
Rewrote the section a bit, lmkwyt.